pkg/dockerregistry/server/auth.go

```diff
@@ -53,7 +53,7 @@ const (
 // RegistryClient encapsulates getting access to the OpenShift API.
 type RegistryClient interface {
 	// Clients return the authenticated clients to use with the server.
-	Clients() (client.Interface, kclientset.Interface, error)
+	Clients() (client.Interface, kcoreclient.LimitRangesGetter, error)
```

Comment:
Why do we need to call `.Core()` anyway? It seems to me like the `kclientset.Interface` is convertible directly to `kcoreclient.CoreInterface`. If not, the `RegistryClient.Clients()` could return the `kcoreclient.CoreInterface` directly until we need some kube client extensions.